session/session.go

```diff
@@ -867,6 +867,10 @@ func (s *session) ExecRestrictedSQLWithContext(ctx context.Context, sql string,
 		}()
 	}
 
+	if execOption.AnalyzeVer != 0 {
+		se.GetSessionVars().AnalyzeVersion = execOption.AnalyzeVer
+	}
+
```

Comment:
If I get it right. we are using the session variable from `autoAnalyzeTable()` to update the session variable here. But from `NewHandle`, the `Handle.mu.ctx` and the `Handle.restrictedExec` should be the same session. So maybe we don't need to update the analyze version here?

Reply:
Here is a little complex. It actually get one session from a session pool to execute. So we need this option to set the analyze version for session which we get from the pool.